ci/playbooks/multinode-customizations.yml

```diff
@@ -9,85 +9,133 @@
-        state: restarted
-        name: crc-pre
+      block:
+        - name: Manage name resolution and interfaces
+          become: true
+          vars:
+            _crc_default_iface: >-
+              {{
+                 _decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.iface
+              }}
+            _crc_default_gw: >-
+              {{
+                 _decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.gw
+              }}
+            _crc_private_connection_name: >-
+              {{
+                 _decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.connection
+              }}
+          block:
+            - name: Ensure crc knows about its second NIC
+              community.general.nmcli:
+                autoconnect: true
+                conn_name: "{{ _crc_private_connection_name }}"
+                dns4: 127.0.0.1
+                ifname: "{{ _crc_default_iface }}"
+                type: ethernet
+                ip4: "{{ _crc_default_net_ip }}"
+                gw4: "{{ _crc_default_gw }}"
+                state: present
 
-    # Avoid 'state: restarted' due to issues with IP not available when crc-dnsmasq starts
-    - name: Stop dnsmasq
-      become: true
-      ansible.builtin.systemd:
-        state: stopped
-        name: crc-dnsmasq
+            - name: Ensure crc does not get "public" DNS
+              become: true
+              community.general.nmcli:
+                autoconnect: true
+                conn_name: "Wired connection 1"
+                dns4_ignore_auto: true
+                state: present
+
+            - name: Restart NetworkManager
+              ansible.builtin.service:
+                name: NetworkManager
+                state: reloaded
+
+        - name: Configure dnsmasq
+          vars:
+            _dnsmasq_config: >-
+              {{
+                _dnsmasq.stat.exists |
+                ternary(_dnsmasq.stat.path, '/etc/dnsmasq.d/crc-dnsmasq.conf')
+              }}
+            _crc_default_net_dns: >-
+              {{
+                 _decoded_net_env.crc_ci_bootstrap_provider_dns |
+                default([])
+              }}
+          block:
+            - name: Configure dns forwarders
+              become: true
+              ansible.builtin.blockinfile:
+                path: "{{ _dnsmasq_config }}"
+                block: |-
+                  {% for dns_server in _crc_default_net_dns %}
+                  server={{ dns_server }}
+                  {% endfor %}
 
-    - name: Make sure that crc-dnsmasq is not running
-      containers.podman.podman_container:
-        name: crc-dnsmasq
-        state: absent
+            - name: Configure local DNS for CRC pod
+              become: true
+              register: last_modification
+              ansible.builtin.replace:
+                path: "{{ _dnsmasq_config }}"
+                regexp: "192.168.130.11"
+                replace: "{{ _crc_default_net_ip }}"
```

Comment:
There is a small change in the behavior here. The old code set there the IP only, without the network prefix length, as the `_decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.ip` value was filtered with `ansible.utils.ipaddr('address')`.

Reply:
oh, good catch. It seems to be working though, but indeed, for the sake of consistency - lemme update it.